### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by philipp94831**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
-    id("net.researchgate.release") version "2.8.1"
+    id("net.researchgate.release") version "3.0.2"
     id("com.bakdata.sonar") version "1.1.11"
     id("com.bakdata.sonatype") version "1.1.11"
-    id("org.hildan.github.changelog") version "0.8.0"
-    id("com.google.cloud.tools.jib") version "3.3.0"
-    id("io.freefair.lombok") version "5.3.3.3"
+    id("org.hildan.github.changelog") version "1.12.1"
+    id("com.google.cloud.tools.jib") version "3.4.0"
+    id("io.freefair.lombok") version "6.6.3"
```
Comment (raminqaf) [inline build.gradle.kts:10]: Why not the `8.4`? If we upgrade to Java 21, version 6 will not be compatible.

File: build.gradle.kts [hunk 2]
```diff
@@ -1,16 +1,13 @@
...
 plugins {
     `java-library`
-    id("net.researchgate.release") version "2.8.1"
+    id("net.researchgate.release") version "3.0.2"
     id("com.bakdata.sonar") version "1.1.11"
     id("com.bakdata.sonatype") version "1.1.11"
-    id("org.hildan.github.changelog") version "0.8.0"
-    id("com.google.cloud.tools.jib") version "3.3.0"
-    id("io.freefair.lombok") version "5.3.3.3"
+    id("org.hildan.github.changelog") version "1.12.1"
```
Comment (raminqaf) [inline build.gradle.kts:8]: From my experience, the upgrade to `2.0.0` went smoothly in other projects.
If you wish to go for a non-major update then let stick with [1.13.1](https://github.com/joffrey-bion/gradle-github-changelog/releases/tag/1.13.1)

File: build.gradle.kts (hunk 1 above)
Comment (philipp94831) [reply build.gradle.kts:10]: One step at a time. We need to check if our plugins work with Gradle 8. I also still need to test this PR, should have marked as draft :)

Comment (philipp94831) [reply build.gradle.kts:10]: I would test Gradle updates in another PR. Fine for you @raminqaf ?

Comment (philipp94831) [reply build.gradle.kts:10]: Tested the PR btw and it works

**Review by raminqaf — APPROVED** [review]
LGTM

**Pull request merged**